extensions/wikia/Forum/ForumBoard.class.php

```diff
@@ -120,12 +77,41 @@ public function getTotalActiveThreads( $relatedPageId = 0, $db = DB_SLAVE ) {
 		return $activeThreads;
 	}
 
+	private function getBoardInfoFromMaster() {
+		$db = wfGetDB( DB_MASTER );
+		$row = $db->selectRow(
+			[ 'comments_index' ],
+			[ 'count(if(parent_comment_id=0,1,null)) threads, count(*) posts, max(first_rev_id) rev_id' ],
+			[
+				'parent_page_id' => $this->getId(),
+				'archived' => 0,
+				'deleted' => 0,
+				'removed' => 0
+			],
+			__METHOD__
+		);
+
+		$forumBoardInfo = new ForumBoardInfo();
+
+		if ( $row ) {
+			$forumBoardInfo->setPostCount( $row->posts );
+			$forumBoardInfo->setThreadCount( $row->threads );
+			$forumBoardInfo->setLastPost( $this->getLastPost( $row->rev_id ) );
+		}
+		$forumBoardInfo->setId( $this->getTitle()->getArticleID() );
+		$forumBoardInfo->setName( $this->getTitle()->getText() );
+		$forumBoardInfo->setUrl( $this->getTitle()->getFullURL() );
+		$forumBoardInfo->setDescription( $this->getDescriptionWithoutTemplates() );
+		$memKey = self::getBoardInfoCacheKey( $this->getId() );
+		$this->wg->Memc->set( $memKey, $forumBoardInfo->toArray(), 60 * 60 * 12 );
+		return $forumBoardInfo;
+	}
+
 	/**
 	 * clear cache for board info
-	 * @param array $parentPageIds
 	 */
 	public function clearCacheBoardInfo() {
-		$this->getBoardInfo( DB_MASTER );
+		$this->getBoardInfoFromMaster();
```

Comment:
personally I don't like situation when the method name is getSomething and instead of using the returned value we are calling it only because of side effects (cache in this example) But TBH I have no better proposition for name for this method